

The crypto type covers scalar (not vector) crypto instructions, and vclz
and vctz appear in two places.

gcc/ChangeLog:

	* config/riscv/generic-vector-ooo.md (vec_crypto): Remove crypto,
	vclz, vctz from insn reservation, rename to vec_pop.
---
 gcc/config/riscv/generic-vector-ooo.md | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gcc/config/riscv/generic-vector-ooo.md b/gcc/config/riscv/generic-vector-ooo.md
index 773003b49ae..bdd64f1a1fb 100644
--- a/gcc/config/riscv/generic-vector-ooo.md
+++ b/gcc/config/riscv/generic-vector-ooo.md
@@ -83,10 +83,10 @@
        (eq_attr "type" "vfmul,vfwmul,vfmuladd,vfwmuladd,vfwmaccbf16,sf_vqmacc,sf_vfnrclip"))
   "vxu_ooo_issue,vxu_ooo_alu")
 
-;; Vector crypto, assumed to be a generic operation for now.
-(define_insn_reservation "vec_crypto" 4
+;; Vector population count
+(define_insn_reservation "vec_pop" 4
   (and (eq_attr "tune" "generic_ooo,generic")
-       (eq_attr "type" "crypto,vclz,vctz,vcpop"))
+       (eq_attr "type" "vcpop"))
   "vxu_ooo_issue,vxu_ooo_alu")
 
 ;; Vector crypto, AES
-- 
2.34.1


